File: model/client/models/book.py
from typing import Optional

from pydantic import BaseModel


class Book(BaseModel):
    id: Optional[str] = None
    title: str
    author: Optional[str] = None
    year: Optional[int] = None
    description: Optional[str] = None
    genre: Optional[list] = None
    cover: Optional[str] = None
    pages: Optional[int] = None

File: mobile/src/types/book.ts
export interface BookEntity {
  id: string;
  title: string;
  author: string;
  year: string;
  description: string;
  genres: string[];
  cover?: string;
  photos?: string[];
  pages: number;
}

export interface PageMetaEntity {
  first: number;
  items: number;
  last: number;
  next: number | null;
  page: number;
  prev: number | null;
}

export interface PagedBooksEntity extends PageMetaEntity {
  data: BookEntity[];
}
